apps/condo/domains/user/integration/identity/SberIdIdentityIntegration.js

```diff
@@ -0,0 +1,177 @@
+const { customAlphabet } = require('nanoid')
+const { isNil } = require('lodash')
+const jwtDecode = require('jwt-decode')
+const axios = require('axios').default
+const https = require('https')
+
+const conf = require('@open-condo/config')
+
+const { SBER_ID_IDP_TYPE } = require('@condo/domains/user/constants/common')
+const { USER_EXTERNAL_IDENTITY_AUTH_CALLBACK_PATH } = require('@condo/domains/user/constants/links')
+const AbstractIdentityIntegration = require('@condo/domains/user/integration/identity/AbstractIdentityIntegration')
+
+// get sber id configuration params
+const SBER_ID_CONFIG = process.env.SBER_ID_CONFIG ? JSON.parse(process.env.SBER_ID_CONFIG) : {}
+const {
+    authorizeUrl,
+    tokenUrl,
+    userInfoUrl,
+    scope,
+    clientId,
+    clientSecret,
+    redirectUri,
+    cert,
+    key,
+    verifyServerSsl,
+} = SBER_ID_CONFIG
+const callbackPath = USER_EXTERNAL_IDENTITY_AUTH_CALLBACK_PATH.replace(':type', SBER_ID_IDP_TYPE)
+const callbackUri = redirectUri || `${conf.SERVER_URL}${callbackPath}`
+const axiosTimeout = 10000
+
+// instantiate httpsAgent in order to support mTLS communication with sber id servers
+const httpsAgent = new https.Agent({
+    rejectUnauthorized: verifyServerSsl,
+    cert,
+    key,
+})
+
+// instantiate request id generator
+const nanoid = customAlphabet('1234567890abcdef', 32)
+
+class SberIdIdentityIntegration extends AbstractIdentityIntegration {
+
+    getType () {
+        return SBER_ID_IDP_TYPE
+    }
+
+    async generateLoginFormParams (checks) {
+        const { nonce, state } = checks
+        const link = new URL(authorizeUrl)
+
+        // generate params
+        const responseType = 'code'
+        const redirectUri = callbackUri
+
+        // set params to link
+        link.searchParams.set('scope', scope)
+        link.searchParams.set('state', state)
+        link.searchParams.set('response_type', responseType)
+        link.searchParams.set('client_id', clientId)
+        link.searchParams.set('redirect_uri', redirectUri)
+        link.searchParams.set('nonce', nonce)
+
+        return link
+    }
+
+    async issueExternalIdentityToken ({ code }) {
+        // set issue token request parameters
+        const request = {
+            grant_type: 'authorization_code',
+            code,
+            redirect_uri: callbackUri,
+            client_id: clientId,
+            client_secret: clientSecret,
+        }
+
+        // send a request
+        const tokenResponse = await axios.create({
```

Comment:
why you don't use the OIDC client?

Reply:
OIDC client doesn't allow to put custom headers into request (this required for sber id auth requests)